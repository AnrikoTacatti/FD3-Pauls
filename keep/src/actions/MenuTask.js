
import api from '../api/api.js';
import { OPEN_FORM_NEW_TASK_CAPTION, OPEN_FORM_EDIT_TASK_CAPTION } from '../stores/const.js';


export const actionopenFormNewTaskCaption = (data, dispatch) => {
    dispatch({ type: OPEN_FORM_NEW_TASK_CAPTION, data });
}
export const actionopenFormEditTaskCaption = (data, dispatch) => {
    dispatch({ type: OPEN_FORM_EDIT_TASK_CAPTION, data });
}
export const actiondeleteChapter = (data, dispatch) => {
    api.removeTaskCaption(data, dispatch);
}

