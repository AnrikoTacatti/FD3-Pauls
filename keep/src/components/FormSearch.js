import React from 'react';
import { connect } from 'react-redux';
import { SEARH_TEXT } from '../stores/const.js';
import { withRouter } from "react-router";
class FormSearch extends React.PureComponent {
    constructor(props) {
        super(props);
    }
    Sarch = () => {

        let data = { text: this.inputSarch.value };
        this.props.dispatch({ type: SEARH_TEXT, data });
        this.props.history.push("/serch");

    }
    render() {
        return (
            <React.Fragment>
                <div className="form-search">
                    <div className="form-inner">
                        <input type="text" className="task-list-name" name="phone" placeholder="Search..." required="" ref={c => this.inputSarch = c} />
                        <button type="button" className="form-submit btn" onClick={this.Sarch} >Sarch</button>
                    </div>
                </div>

            </React.Fragment>
        );

    }

}



export default connect()(withRouter(FormSearch));






