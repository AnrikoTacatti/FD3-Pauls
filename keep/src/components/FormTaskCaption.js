import React from 'react';
import { connect } from 'react-redux';
import {
    actioncloseFormNewTaskCaption,
    actioncloseFormEditTaskCaption,
    actionsetNewTaskCaption,
    actioneditTaskCaption
} from '../actions/FormTaskCaption.js';



export class FormTaskCaption extends React.PureComponent {
    constructor(props) {
        super(props);
        this.state = {
            /* TaskLists: this.props.stateTaskLists.TaskLists*/
            openFormNewTaskCaption: false,
            openFormEditTaskCaption: false
        }
    }

    UNSAFE_componentWillReceiveProps(newProps) {
        this.setState({ openFormNewTaskCaption: newProps.openFormNewTaskCaption });
        this.setState({ openFormEditTaskCaption: newProps.openFormEditTaskCaption });
    }

    closeFormNewTaskCaption = () => {
        let data = false;
        actioncloseFormNewTaskCaption(data, this.props.dispatch)
    }

    closeFormEditTaskCaption = () => {
        let data = { active: false };
        actioncloseFormEditTaskCaption(data, this.props.dispatch);
    }

    addNewTaskCaption = () => {
        let data = { title: this.inputTaskListName.value };
        actionsetNewTaskCaption(data, this.props.dispatch);
    }

    editTaskCaption = () => {
        let data = { title: this.inputTaskListName.value, keychapter: this.state.openFormEditTaskCaption.keychapter };
        actioneditTaskCaption(data, this.props.dispatch);
    }


    render() {
        // console.log("FormTaskCaption", this.props);

        return (
            <React.Fragment>
                {this.state.openFormEditTaskCaption.active === true && <React.Fragment>
                    <div className="form-add-task-caption form">
                        <div className="form-inner">
                            <p className="form-title">Edit task list</p>
                            <input type="text" className="task-list-name" name="title" defaultValue={this.state.openFormEditTaskCaption.name} placeholder="Enter task list name" required="" ref={c => this.inputTaskListName = c} />
                        </div>
                        <div className="form-add-task-caption__footer">
                            <button type="button" title="Cancel" className="form-cancel btn" onClick={this.closeFormEditTaskCaption}>Cancel</button>
                            <button type="button" className="form-submit btn" onClick={this.editTaskCaption} >Submit</button>
                        </div>
                    </div>
                    <div className="form-bg"></div>
                </React.Fragment>
                }
                {this.state.openFormNewTaskCaption === true && <React.Fragment>
                    <div className="form-add-task-caption form">
                        <div className="form-inner">
                            <p className="form-title">Add task list</p>
                            <input type="text" className="task-list-name" name="phone" placeholder="Enter task list name" required="" ref={c => this.inputTaskListName = c} />
                        </div>
                        <div className="form-add-task-caption__footer">
                            <button type="button" title="Cancel" className="form-cancel btn" onClick={this.closeFormNewTaskCaption}>Cancel</button>
                            <button type="button" className="form-submit btn" onClick={this.addNewTaskCaption} >Submit</button>
                        </div>
                    </div>
                    <div className="form-bg"></div>
                </React.Fragment>
                }
            </React.Fragment>
        );

    }

}

const mapStateToProps = function (state) {
    return {
        // весь раздел Redux state под именем counters будет доступен
        // данному компоненту как this.props.counters
        openFormNewTaskCaption: state.stateTaskLists.openFormNewTaskCaption,
        openFormEditTaskCaption: state.stateTaskLists.openFormEditTaskCaption,
    };
};

export default connect(mapStateToProps)(FormTaskCaption);






