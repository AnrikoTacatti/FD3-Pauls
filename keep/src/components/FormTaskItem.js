import React from 'react';
import { connect } from 'react-redux';
import {
    actioncloseFormNew,
    actioncloseFormEdit,
    actionsetNewTaskItem,
    setTaskItem
} from '../actions/FormTaskItem.js';

export class FormTaskItem extends React.PureComponent {
    constructor(props) {
        super(props);
        this.state = {
            openFormTaskItemEdit: false,
            openFormTaskItemNew: false,
            inputTaskItemNameRequired: true,
            inputTaskItemTextRequired: true,
        };
        this.inputTaskItemName = React.createRef();
        this.inputTaskItemText = React.createRef();

    }

    UNSAFE_componentWillReceiveProps(newProps) {
        this.setState({ openFormTaskItemEdit: newProps.openFormTaskItemEdit });
        this.setState({ openFormTaskItemNew: newProps.openFormTaskItemNew });

    }

    closeFormNew = () => {
        let data = { active: false };
        actioncloseFormNew(data, this.props.dispatch);
    }


    closeFormEdit = () => {
        let data = { active: false };
        actioncloseFormEdit(data, this.props.dispatch);
    }

    addNewTaskItem = () => {
        if (this.inputTaskItemName.current.value == "") {
            this.setState({ inputTaskItemNameRequired: false });
        }
        else {
            this.setState({ inputTaskItemNameRequired: true });
        }
        if (this.inputTaskItemText.current.value == "") {
            this.setState({ inputTaskItemTextRequired: false });
        }
        else {
            this.setState({ inputTaskItemTextRequired: true });
        }
        if (this.inputTaskItemName.current.value != "" && this.inputTaskItemText.current.value != "") {
            let data = { keychapter: this.state.openFormTaskItemNew.keychapter, title: this.inputTaskItemName.current.value, text: this.inputTaskItemText.current.value };
            actionsetNewTaskItem(data, this.props.dispatch);
        }
    }

    editTaskItem = () => {
        let data = { title: this.inputTaskItemName.value, text: this.inputTaskItemText.value, keychapter: this.state.openFormTaskItemEdit.keychapter, keyitem: this.state.openFormTaskItemEdit.keyitem };
        setTaskItem(data, this.props.dispatch);
    }

    render() {
        //  console.log("FormTaskItem", this.props);
        // console.log("FormTaskItem", this.state);
        return (
            <React.Fragment >
                {
                    this.state.openFormTaskItemNew.activ === true && <React.Fragment>
                        <div className="form-add-task-caption form">
                            <div className="form-inner">
                                <p className="form-title">Add task item</p>
                                <input type="text" className={`task-item-name ${this.state.inputTaskItemNameRequired == false ? "error" : ""}`} name="title" placeholder="Enter task item name" required="" ref={this.inputTaskItemName} />
                                <textarea className={`task-item-text ${this.state.inputTaskItemTextRequired == false ? "error" : ""}`} name="text" cols="60" rows="5" placeholder="Enter task item text" required="" ref={this.inputTaskItemText} />
                            </div>
                            <div className="form-add-task-caption__footer">
                                <button type="button" title="Cancel" className="form-cancel btn" onClick={this.closeFormNew}>Cancel</button>
                                <button type="button" className="form-submit btn" onClick={this.addNewTaskItem} >Submit</button>
                            </div>
                        </div>
                        <div className="form-bg"></div>
                    </React.Fragment>
                }
                {
                    this.state.openFormTaskItemEdit.activ === true && <React.Fragment>
                        <div className="form-add-task-caption form">
                            <div className="form-inner">
                                <p className="form-title">Edit task item</p>
                                <input type="text" className="task-item-name" defaultValue={this.state.openFormTaskItemEdit.name} name="name" placeholder="Enter task item name" required="" ref={c => this.inputTaskItemName = c} />
                                <textarea className="task-item-text" defaultValue={this.state.openFormTaskItemEdit.text} name="text" cols="60" rows="5" placeholder="Enter task item text" required="" ref={c => this.inputTaskItemText = c} />
                            </div>
                            <div className="form-add-task-caption__footer">
                                <button type="button" title="Cancel" className="form-cancel btn" onClick={this.closeFormEdit}>Cancel</button>
                                <button type="button" className="form-submit btn" onClick={this.editTaskItem} >Submit</button>
                            </div>
                        </div>
                        <div className="form-bg"></div>
                    </React.Fragment>
                }
            </React.Fragment >
        );

    }

}

const mapStateToProps = function (state) {
    return {
        // весь раздел Redux state под именем counters будет доступен
        // данному компоненту как this.props.counters
        openFormTaskItemEdit: state.stateTaskLists.openFormTaskItemEdit,
        openFormTaskItemNew: state.stateTaskLists.openFormTaskItemNew
    };
};

export default connect(mapStateToProps)(FormTaskItem);






