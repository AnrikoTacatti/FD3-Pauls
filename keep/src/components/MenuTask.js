import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';


class MenuTask extends React.Component {
    constructor(props) {
        super(props);
        this.state = {}
    }

    folder = () => <svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="folder" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512" ><path fill="currentColor" d="M464 128H272l-64-64H48C21.49 64 0 85.49 0 112v288c0 26.51 21.49 48 48 48h416c26.51 0 48-21.49 48-48V176c0-26.51-21.49-48-48-48z" ></path></svg>;

    render() {
        console.log("MenuTask ");
        console.log(this.props.TaskLists);
        return (

            <React.Fragment>
                {/*this.props.TaskLists.map((val) => <li><NavLink to="/" exact className="PageLink" activeClassName="ActivePageLink"> {this.folder() + " " + val.itemlist.name} </NavLink></li>

        )*/}
            </React.Fragment>
        );

    }

}

const mapStateToProps = function (state) {
    return {
        // весь раздел Redux state под именем counters будет доступен
        // данному компоненту как this.props.counters
        TaskLists: state.TaskLists,
    };
};

export default connect(mapStateToProps)(MenuTask);






