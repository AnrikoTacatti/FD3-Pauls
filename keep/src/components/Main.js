"use strict";
import React from 'react';
import FormTaskItem from './FormTaskItem.js';
import TaskItemContainer from './TaskItemContainer.js';
import { icoSave } from '../ico/ico.js';
import { actionsaveBook } from '../actions/Main.js';
class Main extends React.PureComponent {
    constructor(props) {
        super(props);
    }

    forTaskListschild = () => {
        let tasklistitem = [];
        let listtask = this.props.TaskListsItemsSort;
        let length = 0;

        switch (true) {
            case listtask.length >= 4: length = 4; break;
            case listtask.length >= 3: length = 3; break;
            case listtask.length >= 2: length = 2; break;
            case listtask.length >= 1: length = 1; break;
        }

        for (let i = 0; i < length; i++) {
            let keyitem = listtask[i].key;
            let keychapter = listtask[i].keychapter;
            tasklistitem.push(
                <TaskItemContainer data={listtask[i]} keychapter={keychapter} keyitem={keyitem} key={keyitem} index={i + 1}
                    attrdata={this.props.locationPathname !== undefined ? this.props.locationPathname : "all"}
                />
            )
        }

        return tasklistitem;
    }

    forPinTaskListschild = () => {
        let tasklistitem = [];
        let listtask = [];
        if (this.props.TaskListsItemsSort.length) this.props.TaskListsItemsSort.forEach(function (el, index, array) { el.pin === true && listtask.push(el) });
        let length = 0;

        switch (true) {
            case listtask.length >= 4: length = 4; break;
            case listtask.length >= 3: length = 3; break;
            case listtask.length >= 2: length = 2; break;
            case listtask.length >= 1: length = 1; break;
        }

        for (let i = 0; i < length; i++) {
            let keyitem = listtask[i].key;
            let keychapter = listtask[i].keychapter;
            tasklistitem.push(
                <TaskItemContainer data={listtask[i]} keychapter={keychapter} keyitem={keyitem} key={keyitem} index={i + 1}
                    attrdata={this.props.locationPathname !== undefined ? this.props.locationPathname : "all"}
                />
            )
        }
        return tasklistitem;
    }


    render() {
        console.log("render Main");
        return (
            <React.Fragment>
                <div className="task-chapter task-chapter--main" >
                    <style>
                        {` .task-item{ -webkit-animation-name: none;
                                animation-name: none;} 
                            .${this.props.locationPathname !== undefined ? this.props.locationPathname : "all"} { 
                                -webkit-animation-name: slideInUp;
                                animation-name: slideInUp;
                                opacity: 0; }
                            `
                        }
                    </style>
                    <div className="task-chapter__header">
                        <div className="task-chapter__title">
                            Последние задачи
                        </div>
                    </div>
                    <div className="task-item-list">
                        {this.forTaskListschild()}
                    </div>
                    <div className="task-chapter__header">
                        <div className="task-chapter__title">
                            Закрепленные задачи
                        </div>
                    </div>
                    <div className="task-item-list">
                        {this.forPinTaskListschild()}
                    </div>
                    <div className="task-chapter__header">
                        <div className="task-chapter__title">
                            Записная книжка
                        </div>
                    </div>
                    <div className="book" style={{ "animationDelay": `1s` }}>
                        <div className="book__title">Записная книжка <span className="book__save" onClick={this.props.cbsaveBook}> {icoSave()} </span></div>
                        {console.log("Записная книжка")}
                        {this.props.Book !== undefined && <textarea className="book__text" ref={this.props.cbsetNewTextRef} defaultValue={this.props.Book !== undefined && this.props.Book.text} placeholder="Запишите что-нибудь"></textarea>}
                    </div>

                </div>
                <FormTaskItem />
            </React.Fragment>

        );
    }


}

export default Main;
