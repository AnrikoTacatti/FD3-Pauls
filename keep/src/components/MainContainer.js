"use strict";
import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from "react-router";
import FormTaskItem from './FormTaskItem.js';
import Main from './Main.js';
import api from '../api/api.js';

class MainContainer extends React.PureComponent {
    constructor(props) {
        super(props);
        this.state = {
            TaskListsItemsSort: this.props.stateTaskListsItemsSort,
            locationPathname: this.props.match.params.chapter,
            Book: this.props.Book
        }

    }

    UNSAFE_componentWillReceiveProps = (newProps) => {
        this.setState({ TaskListsItemsSort: newProps.stateTaskListsItemsSort });
        this.setState({ locationPathname: newProps.match.params.chapter });
        this.setState({ Book: newProps.Book });
    }

    saveBook = () => {
        let data = { text: this.bookText.value };
        api.saveBook(data, this.props.dispatch);

    }
    setNewTextRef = (ref) => {
        this.bookText = ref;
    };

    render() {
        console.log("render MainContainer");
        return (
            <React.Fragment>
                <Main {...this.state} cbsetNewTextRef={this.setNewTextRef} cbsaveBook={this.saveBook} />
                <FormTaskItem />
            </React.Fragment>
        );
    }


}

const mapStateToProps = function (state) {
    return {
        // весь раздел Redux state под именем counters будет доступен
        // данному компоненту как this.props.counters
        stateTaskListsItemsSort: state.stateTaskLists.TaskListsItemsSort,
        Book: state.stateTaskLists.Book
    };
};

export default connect(mapStateToProps)(withRouter(MainContainer));
