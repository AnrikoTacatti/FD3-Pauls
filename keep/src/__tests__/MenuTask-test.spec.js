"use strict";
import React from 'react';
import { NavLink } from 'react-router-dom';
import { MenuTask } from '../components/MenuTask';
jest.mock('../actions/MenuTask.js');
import {
    actionopenFormNewTaskCaption,
    actionopenFormEditTaskCaption,
    actiondeleteChapter

} from '../actions/MenuTask.js';

let taskLists = {
    "0-test": {
        "itemlist": {
            "-Mnu6P7dMJTeVv8VESLh": {
                "name": "Мои заметки № 1",
                "text": "Мои заметки № 1",
                "time": 1636283818894,
                "key": "-Mnu6P7dMJTeVv8VESLh"
            },
            "-Mnu6RA2cuqhZiUml2eL": {
                "name": "Мои заметки № 2",
                "pin": true,
                "text": "Мои заметки № 2",
                "time": 1636283827275,
                "key": "-Mnu6RA2cuqhZiUml2eL"
            }
        },
        "name": "Мои заметки",
        "url": "moi-zametki",
        "key": "0-test"
    },
    "1-test": {
        "itemlist": {
            "-Mnu6lwJvGFmsBIjhPBh": {
                "name": "Заметки работа №1",
                "text": "Заметки работа №1",
                "time": 1636283916422,
                "key": "-Mnu6lwJvGFmsBIjhPBh"
            },
            "-Mnu6n_j_6Szfb5fRtuR": {
                "name": "Заметки работа №2",
                "text": "Заметки работа №2",
                "time": 1636283923175,
                "key": "-Mnu6n_j_6Szfb5fRtuR"
            }
        },
        "name": "Заметки работа",
        "url": "zametki-rabota",
        "key": "1-test"
    },
    "-MnosmgmdFWwUg_Qyj5o": {
        "itemlist": {
            "-Mnu71_BYp5YokQlvrg4": {
                "name": "Заметки учеба №111",
                "pin": true,
                "style": {
                    "backgroundColor": "#ffecf1",
                    "border": "1px solid #f9c8ca"
                },
                "text": "Заметки учеба №1",
                "time": 1636466420836,
                "key": "-Mnu71_BYp5YokQlvrg4"
            },
            "-Mnu736Ol-_bA2P_gxQg": {
                "name": "Заметки учеба №2",
                "text": "Заметки учеба №2",
                "time": 1636283990848,
                "key": "-Mnu736Ol-_bA2P_gxQg"
            },
            "-Mnu74Jhfq6DAtIDT796": {
                "name": "Заметки учеба №3",
                "text": "",
                "time": 1636465260116,
                "key": "-Mnu74Jhfq6DAtIDT796-test"
            }
        },
        "name": "Заметки учеба",
        "time": 1636196167296,
        "url": "zametki-ucheba",
        "key": "-MnosmgmdFWwUg_Qyj5o"
    },
    "-Mnot8F4fnIev0gGlnxg": {
        "itemlist": {
            "-Mnu7QdYFIdU_yM1_eCE": {
                "name": "Заметки хобби № 1",
                "text": "Заметки хобби № 1",
                "time": 1636284087310,
                "key": "-Mnu7QdYFIdU_yM1_eCE"
            },
            "-Mnu7SIorvdewdoTTefa": {
                "name": "Заметки хобби № 2",
                "text": "Заметки хобби № 2",
                "time": 1636284094021,
                "key": "-Mnu7SIorvdewdoTTefa"
            }
        },
        "name": "Заметки хобби",
        "time": 1636196193514,
        "url": "zametki-hobbi",
        "key": "-Mnot8F4fnIev0gGlnxg-test"
    }
};



describe('TaskChapter-test', () => {
    let state = {
        TaskLists: taskLists,
    }
    let dispatch = jest.fn();
    actionopenFormNewTaskCaption.mockImplementation((dispatch, data) => { return data; });
    actionopenFormEditTaskCaption.mockImplementation((dispatch, data) => { return data; });
    actiondeleteChapter.mockImplementation((dispatch, data) => { return data; });
    let UNSAFE_componentWillReceiveProps = jest.spyOn(MenuTask.prototype, "UNSAFE_componentWillReceiveProps");

    const component = shallow(
        <MenuTask  {...state} dispatch={dispatch} />
    );

    test('MenuTask Snapshot zametki-ucheba', () => {
        expect(component).toMatchSnapshot();
    });
    test('MenuTask  New UNSAFE_componentWillReceiveProps', () => {
        component.setProps({
            TaskLists: taskLists,
        });
        expect(UNSAFE_componentWillReceiveProps).toHaveBeenCalled();
    });

    test('MenuTask  TaskItemContainer length', () => {
        const TaskItemContainerobj = component.find(NavLink);
        expect(TaskItemContainerobj).toHaveLength(6);
    });
    test('MenuTask openFormEditTaskCaption', () => {
        const btn = component.find("li").at(2).find(".openFormEditTaskCaption");
        btn.simulate('click');
        expect(actionopenFormEditTaskCaption).toHaveBeenCalledWith({ active: true, keychapter: "0-test", name: "Мои заметки" }, dispatch);
    });
    test('MenuTask actiondeleteChapter', () => {
        let confirm = jest.spyOn(window, 'confirm');
        confirm.mockImplementation(jest.fn(() => true));
        const btn = component.find("li").at(2).find(".deleteChapter");
        btn.simulate('click');
        expect(actiondeleteChapter).toHaveBeenCalledWith({ keychapter: "0-test" }, dispatch);
    });

    test('MenuTask openFormNewTaskCaption', () => {
        const btn = component.find(".openFormNewTaskCaption");
        btn.simulate('click');
        expect(actionopenFormNewTaskCaption).toHaveBeenCalledWith(true, dispatch);
    });
});








