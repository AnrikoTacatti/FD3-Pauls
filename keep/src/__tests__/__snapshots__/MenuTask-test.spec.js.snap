// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskChapter-test MenuTask Snapshot zametki-ucheba 1`] = `
<Fragment>
  <li>
    <NavLink
      activeClassName="ActivePageLink"
      className="PageLink"
      exact={true}
      to="/chapter"
    >
       
      <svg
        aria-hidden="true"
        data-icon="folder"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M464 128H272l-64-64H48C21.49 64 0 85.49 0 112v288c0 26.51 21.49 48 48 48h416c26.51 0 48-21.49 48-48V176c0-26.51-21.49-48-48-48z"
          fill="currentColor"
        />
      </svg>
       Все 
    </NavLink>
  </li>
  <li>
    <NavLink
      activeClassName="ActivePageLink"
      className="PageLink"
      exact={true}
      to="/pin"
    >
       
      <svg
        aria-hidden="true"
        data-icon="thumbtack"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 384 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M298.028 214.267L285.793 96H328c13.255 0 24-10.745 24-24V24c0-13.255-10.745-24-24-24H56C42.745 0 32 10.745 32 24v48c0 13.255 10.745 24 24 24h42.207L85.972 214.267C37.465 236.82 0 277.261 0 328c0 13.255 10.745 24 24 24h136v104.007c0 1.242.289 2.467.845 3.578l24 48c2.941 5.882 11.364 5.893 14.311 0l24-48a8.008 8.008 0 0 0 .845-3.578V352h136c13.255 0 24-10.745 24-24-.001-51.183-37.983-91.42-85.973-113.733z"
          fill="currentColor"
        />
      </svg>
       Закрепленные 
    </NavLink>
  </li>
  <li
    key="0-test"
  >
    <NavLink
      activeClassName="ActivePageLink"
      className="PageLink"
      exact={true}
      key="0-test"
      to="/chapter/moi-zametki"
    >
      <svg
        aria-hidden="true"
        data-icon="folder"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M464 128H272l-64-64H48C21.49 64 0 85.49 0 112v288c0 26.51 21.49 48 48 48h416c26.51 0 48-21.49 48-48V176c0-26.51-21.49-48-48-48z"
          fill="currentColor"
        />
      </svg>
        
      Мои заметки
    </NavLink>
    <span
      className="openFormEditTaskCaption"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="ico-edit"
        data-icon="pen"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M290.74 93.24l128.02 128.02-277.99 277.99-114.14 12.6C11.35 513.54-1.56 500.62.14 485.34l12.7-114.22 277.9-277.88zm207.2-19.06l-60.11-60.11c-18.75-18.75-49.16-18.75-67.91 0l-56.55 56.55 128.02 128.02 56.55-56.55c18.75-18.76 18.75-49.16 0-67.91z"
          fill="currentColor"
        />
      </svg>
    </span>
    <span
      className="deleteChapter"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="ico-trash"
        data-icon="trash"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
          fill="currentColor"
        />
      </svg>
    </span>
  </li>
  <li
    key="1-test"
  >
    <NavLink
      activeClassName="ActivePageLink"
      className="PageLink"
      exact={true}
      key="1-test"
      to="/chapter/zametki-rabota"
    >
      <svg
        aria-hidden="true"
        data-icon="folder"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M464 128H272l-64-64H48C21.49 64 0 85.49 0 112v288c0 26.51 21.49 48 48 48h416c26.51 0 48-21.49 48-48V176c0-26.51-21.49-48-48-48z"
          fill="currentColor"
        />
      </svg>
        
      Заметки работа
    </NavLink>
    <span
      className="openFormEditTaskCaption"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="ico-edit"
        data-icon="pen"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M290.74 93.24l128.02 128.02-277.99 277.99-114.14 12.6C11.35 513.54-1.56 500.62.14 485.34l12.7-114.22 277.9-277.88zm207.2-19.06l-60.11-60.11c-18.75-18.75-49.16-18.75-67.91 0l-56.55 56.55 128.02 128.02 56.55-56.55c18.75-18.76 18.75-49.16 0-67.91z"
          fill="currentColor"
        />
      </svg>
    </span>
    <span
      className="deleteChapter"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="ico-trash"
        data-icon="trash"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
          fill="currentColor"
        />
      </svg>
    </span>
  </li>
  <li
    key="-MnosmgmdFWwUg_Qyj5o"
  >
    <NavLink
      activeClassName="ActivePageLink"
      className="PageLink"
      exact={true}
      key="-MnosmgmdFWwUg_Qyj5o"
      to="/chapter/zametki-ucheba"
    >
      <svg
        aria-hidden="true"
        data-icon="folder"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M464 128H272l-64-64H48C21.49 64 0 85.49 0 112v288c0 26.51 21.49 48 48 48h416c26.51 0 48-21.49 48-48V176c0-26.51-21.49-48-48-48z"
          fill="currentColor"
        />
      </svg>
        
      Заметки учеба
    </NavLink>
    <span
      className="openFormEditTaskCaption"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="ico-edit"
        data-icon="pen"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M290.74 93.24l128.02 128.02-277.99 277.99-114.14 12.6C11.35 513.54-1.56 500.62.14 485.34l12.7-114.22 277.9-277.88zm207.2-19.06l-60.11-60.11c-18.75-18.75-49.16-18.75-67.91 0l-56.55 56.55 128.02 128.02 56.55-56.55c18.75-18.76 18.75-49.16 0-67.91z"
          fill="currentColor"
        />
      </svg>
    </span>
    <span
      className="deleteChapter"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="ico-trash"
        data-icon="trash"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
          fill="currentColor"
        />
      </svg>
    </span>
  </li>
  <li
    key="-Mnot8F4fnIev0gGlnxg"
  >
    <NavLink
      activeClassName="ActivePageLink"
      className="PageLink"
      exact={true}
      key="-Mnot8F4fnIev0gGlnxg"
      to="/chapter/zametki-hobbi"
    >
      <svg
        aria-hidden="true"
        data-icon="folder"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M464 128H272l-64-64H48C21.49 64 0 85.49 0 112v288c0 26.51 21.49 48 48 48h416c26.51 0 48-21.49 48-48V176c0-26.51-21.49-48-48-48z"
          fill="currentColor"
        />
      </svg>
        
      Заметки хобби
    </NavLink>
    <span
      className="openFormEditTaskCaption"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="ico-edit"
        data-icon="pen"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M290.74 93.24l128.02 128.02-277.99 277.99-114.14 12.6C11.35 513.54-1.56 500.62.14 485.34l12.7-114.22 277.9-277.88zm207.2-19.06l-60.11-60.11c-18.75-18.75-49.16-18.75-67.91 0l-56.55 56.55 128.02 128.02 56.55-56.55c18.75-18.76 18.75-49.16 0-67.91z"
          fill="currentColor"
        />
      </svg>
    </span>
    <span
      className="deleteChapter"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="ico-trash"
        data-icon="trash"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
          fill="currentColor"
        />
      </svg>
    </span>
  </li>
  <li>
    <span
      className="openFormNewTaskCaption"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        data-icon="plus"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          fill="currentColor"
        />
      </svg>
       Create new list
    </span>
  </li>
</Fragment>
`;
